streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -328,13 +325,44 @@ public void schedule(long interval) {
         punctuationQueue.schedule(new PunctuationSchedule(currNode, interval));
     }
 
+    @Override
+    public void init() {
+        if (prevPartitionQueues.size() > 0) {
+            partitionQueues.clear();
+            partitionQueues.putAll(prevPartitionQueues);
+            this.partitionGroup = new PartitionGroup(partitionQueues, timestampExtractor);
+            prevPartitionQueues.clear();
+        }
+        if (prevConsumedOffsets.size() > 0) {
```

Comment:
Why do we want to re-create the `partitionGroup` with the previous partition queues? Could we just keep the old `partitionGroup` of the task as is? And also I think we do not need to keep `prevConsumedOffsets` as the `consumedOffsets` can always be initialized by the prev committed offsets which we will still execute for pausing the task.